Automattic/woocommerce-payments
# Update Name Your Price compatibility to use new Compatibility methods

Fixes #5524 #7172

#### Changes proposed in this Pull Request

This PR is the second part of #7202 that uses the new methods in order to allow Name Your Price amounts to be able to be switched on the cart and checkout pages.

Adds param to the "Edit price" link for a NYP product in the cart. When clicked the user is redirected back to the product page with the current price pre-filled. If the user switches the currency at this point the pre-filled price will not change until this patch is implemented.

#### Testing instructions

* Name Your Price can be obtained in the all-plugins repo.
* If you do not have more than one currency in your store, then navigate to WooCommerce > Settings > Multi-Currency and add another currency.
* Create simple NYP product.
* View the product, add an amount, and add it to the cart.
* Go to cart and click on edit link, should redirect to single product page with price pre-filled.
* Use the currency switcher to change currencies, the amount in the NYP field should update accordingly.

-------------------
